0-common/igluctl/src/main/scala/com.snowplowanalytics.iglu/ctl/LintCommand.scala

```diff
@@ -46,11 +46,38 @@ case class LintCommand(inputDir: File, skipWarnings: Boolean, severityLevel: Sev
    */
   def process(): Unit = {
     val jsons = getJsonFilesStream(inputDir, Some(filterJsonSchemas))
-    val reports = jsons.map { file =>
-      val report = file.map(check)
-      flattenReport(report)
+
+    val (failures, validatedJsons) = splitValidations(jsons.toList)
+    if (failures.nonEmpty) {
+      println("JSON Parsing errors:")
+      println(failures.mkString("\n"))
+    }
+
+    // lint schema versions
+    val stubFile: File = new File(inputDir.getAbsolutePath)
+    val stubCommand = GenerateCommand(stubFile, stubFile)
+    val (_, schemas) = splitValidations(validatedJsons.map(_.extractSelfDescribingSchema))
+    val (schemaVerWarnings, schemaVerErrors) = stubCommand.validateSchemaVersions(schemas)
+    val lintSchemaVer: Boolean = (schemaVerWarnings, schemaVerErrors) match {
+      case (Nil, Nil) => true
+      case (warnings, Nil) =>
+        println(warnings.mkString("\n"))
+        true
+      case (Nil, errors) =>
+        println(errors.mkString("\n"))
+        false
+      case (List(_), List(_)) => true
```

Comment:
For me this last match means `if we have one warning and one error then everything is correct`.

Reply:
@chuwy , actually this is the case that never gets executed but inserted here to be able to compile. [validateSchemaVersions](https://github.com/snowplow/iglu/commit/5af7d4d0f35337e1880916b817549fce200493a7#diff-a1f8bc1418c29990fde529ec83cd1691R135) never return warnings together with errors, it is either (no warning, no error), (warning, no error) or (no warning, error) per [ issue strategy](https://github.com/snowplow/iglu/issues/221#issuecomment-353962880).

Reply:
Two points here:

1. If you need a universal fallback case, you write it as `case _ => true`
2. It again looks like a way to trick the compiler and developer who doesn't know about context. In languages with strong type discipline like Scala, one of the most important purposes of types is to provide developer enough information to reason about code using smallest possible scope, ideally just types. Therefore it's easier to spot bugs and test the code. Here developer needs to look into different functions/modules and examine this code to understand that we have a line that never gets executed. That's okay for now, but I'd like to see this code refactored someday (of course, it was me who started this mess, but I was young back then!).

Reply:
Adding a short comment for now.